File: 2008-maximum-earnings-from-taxi/2008-maximum-earnings-from-taxi.cpp
class Solution {
public:
   long long maxTaxiEarnings(int n, vector<vector<int>>& A) {   
       sort(A.begin(), A.end());
       for(int i=0;i<A.size();++i)
       {
           cout<<A[i][0]<<" "<<A[i][1]<<" "<<A[i][2]<<" ";
       }
        vector<long long> dp(n+1);
        int j = 0;
        for(int i = 1; i <= n; ++i) {
            dp[i] = max(dp[i], dp[i - 1]);
            while (j < A.size() && A[j][0] == i)
                dp[A[j++][1]] = max(dp[A[j][1]], dp[i] + A[j][1] - A[j][0] + A[j][2]);
        }
        return dp[n];
    }
};

File: 2008-maximum-earnings-from-taxi/2008-maximum-earnings-from-taxi.java
class Solution {
  public long maxTaxiEarnings(int n, int[][] A) {
        Arrays.sort(A, (a, b) -> a[0] - b[0]);
        long[] dp = new long[n + 1];
        int j = 0;
        for(int i = 1; i <= n; ++i) {
            dp[i] = Math.max(dp[i], dp[i - 1]);
            while (j < A.length && A[j][0] == i) {
                dp[A[j][1]] = Math.max(dp[A[j][1]], dp[i] + A[j][1] - A[j][0] + A[j][2]);
                ++j;
            }
        }
        return dp[n];
    }
}